### DISCUSSION TITLE
rawQuery and enum comparison

### DISCUSSION THREAD

**Discussion by nephix** [prisma/prisma#14471] [Q&A] [answered]
Hey,

I'm having trouble using `rawQuery` and using and enum in the WHERE clause on a postgres db.

Schema:

```
enum Event {
  SIGN_UP
}

model AnalyticsEvent {
  id            Int      @id @default(autoincrement())
  createdAt     DateTime @default(now())
  event         Event
}
```

And the query:

```typescript
import { Event } from '@generated';
const event = Event.SIGN_UP;

await prisma.$rawQuery`
SELECT
      *
FROM
      "AnalyticsEvent"
WHERE
      event = ${event}
`;
```

This always results in:
> Raw query failed. Code: `42883`. Message: `db error: ERROR: operator does not exist: "Event" = text
> HINT: No operator matches the given name and argument types. You might need to add explicit type casts.`

I already tried to cast the `${event}` to `:: TEXT` but I keep getting the same error.

**Comment by nephix** [comment, marked as answer]
The solution is to cast it to the `"Event"` type:

```sql
await prisma.$rawQuery`
SELECT
      *
FROM
      "AnalyticsEvent"
WHERE
      event = ${event}::"Event"
`;
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
